parquet-hadoop/src/main/java/org/apache/parquet/hadoop/rewrite/ParquetRewriter.java

```diff
@@ -904,4 +953,171 @@ public byte[] getDictPageAAD() {
+              .orElseThrow(() -> new IllegalStateException(
+                  "Could not find column [" + columnPath.toDotString() + "]."));
+          int bloomFilterLength = chunk.getBloomFilterLength();
+          ParquetProperties.WriterVersion writerVersion =
+              chunk.getEncodingStats().usesV2Pages()
+                  ? ParquetProperties.WriterVersion.PARQUET_2_0
+                  : ParquetProperties.WriterVersion.PARQUET_1_0;
+          ParquetProperties props = ParquetProperties.builder()
+              .withWriterVersion(writerVersion)
+              .withBloomFilterEnabled(bloomFilterLength > 0)
+              .build();
+          CodecFactory codecFactory = new CodecFactory(new Configuration(), props.getPageSizeThreshold());
+          CompressionCodecFactory.BytesInputCompressor compressor =
+              codecFactory.getCompressor(chunk.getCodec());
+
+          MessageType columnSchema = parquetRewriter.newSchema(schema, descriptor);
+          ColumnChunkPageWriteStore cPageStore = new ColumnChunkPageWriteStore(
+              compressor,
+              columnSchema,
+              props.getAllocator(),
+              props.getColumnIndexTruncateLength(),
+              props.getPageWriteChecksumEnabled(),
+              writer.getEncryptor(),
+              rowGroupIdxIn);
+          ColumnWriteStore cwStore = props.newColumnWriteStore(columnSchema, cPageStore, cPageStore);
+          ColumnWriter cWriter = cwStore.getColumnWriter(descriptor);
+          cPageStores.put(descriptor, cPageStore);
+          cStores.put(descriptor, cwStore);
+          cWriters.put(descriptor, cWriter);
+        });
+      }
+    }
+
+    private void initReaders() throws IOException {
+      if (!inputFiles.isEmpty()) {
+        TransParquetFileReader reader = inputFiles.peek();
+        PageReadStore pageReadStore = reader.readRowGroup(rowGroupIdxOut);
+        String createdBy = reader.getFooter().getFileMetaData().getCreatedBy();
+        ColumnReadStoreImpl crStore = new ColumnReadStoreImpl(
+            pageReadStore, new ParquetRewriter.DummyGroupConverter(), schema, createdBy);
+        for (ColumnDescriptor descriptor : descriptorsMap.values()) {
+          ColumnReader cReader = crStore.getColumnReader(descriptor);
+          colReaders.put(descriptor, cReader);
+        }
+      }
+    }
+
+    private void copyValues(ColumnDescriptor descriptor, long rowsToWrite) {
+      ColumnWriteStore cStore = cStores.get(descriptor);
+      ColumnWriter cWriter = cWriters.get(descriptor);
+      int dMax = descriptor.getMaxDefinitionLevel();
+      Class<?> columnType = descriptor.getPrimitiveType().getPrimitiveTypeName().javaType;
+      ColumnReader reader = colReaders.get(descriptor);
+      for (int i = 0; i < rowsToWrite; i++) {
+        int rlvl = reader.getCurrentRepetitionLevel();
+        int dlvl = reader.getCurrentDefinitionLevel();
+        do {
+          if (dlvl < dMax) {
+            cWriter.writeNull(rlvl, dlvl);
```

Comment:
If a column to mask (only nullifying column is supported) is on the right side, we also need to handle it here.